core/src/main/java/jenkins/telemetry/impl/AutoRefresh.java

```diff
@@ -0,0 +1,113 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package jenkins.telemetry.impl;
+
+import hudson.Extension;
+import jenkins.telemetry.Telemetry;
+import net.sf.json.JSONObject;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.annotation.Nonnull;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpSession;
+import java.time.LocalDate;
+import java.util.HashSet;
+import java.util.Set;
+
+@Extension
+@Restricted(NoExternalUse.class)
+public class AutoRefresh extends Telemetry {
+
+    private static final Set<String> sessionsWithAutoRefresh = new HashSet<>();
+    private static final Set<String> sessionsWithoutAutoRefresh = new HashSet<>();
+
+    @Nonnull
+    @Override
+    public String getId() {
+        return AutoRefresh.class.getName();
+    }
+
+    @Nonnull
+    @Override
+    public String getDisplayName() {
+        return Messages.AutoRefresh_DisplayName();
+    }
+
+    @Nonnull
+    @Override
+    public LocalDate getStart() {
+        return LocalDate.of(2019, 10, 20);
+    }
+
+    @Nonnull
+    @Override
+    public LocalDate getEnd() {
+        return LocalDate.of(2019, 12, 31);
+    }
+
+    @Override
+    public JSONObject createContent() {
+        int sessionsWithAutoRefreshCount, sessionsWithoutAutoRefreshCount;
+        synchronized (sessionsWithAutoRefresh) {
+            sessionsWithAutoRefreshCount = sessionsWithAutoRefresh.size();
+            sessionsWithAutoRefresh.clear();
+        }
+        synchronized (sessionsWithoutAutoRefresh) {
+            sessionsWithoutAutoRefreshCount = sessionsWithoutAutoRefresh.size();
+            sessionsWithoutAutoRefresh.clear();
+        }
+
+        if (sessionsWithAutoRefreshCount == 0 && sessionsWithoutAutoRefreshCount == 0) {
+            return null;
+        }
+
+        JSONObject payload = new JSONObject();
+        payload.put("enabled", sessionsWithAutoRefreshCount);
+        payload.put("disabled", sessionsWithoutAutoRefreshCount);
+        return payload;
+    }
+
+    public void recordRequest(HttpServletRequest request, boolean enabled) {
+        if (Telemetry.isDisabled()) {
+            return;
+        }
+        if (!isActivePeriod()) {
+            return;
+        }
+        HttpSession session = request.getSession(false);
+        if (session != null) {
+            String sessionId = session.getId();
+            if (enabled) {
+                synchronized (sessionsWithAutoRefresh) {
+                    sessionsWithAutoRefresh.add(sessionId);
+                }
+            } else {
+                synchronized (sessionsWithoutAutoRefresh) {
+                    sessionsWithoutAutoRefresh.add(sessionId);
+                }
+            }
```

Comment:
Maybe overkill but:

```suggestion
            Set<String> sessions = enabled ? sessionsWithAutoRefresh : sessionsWithoutAutoRefresh;
            synchronized (sessions) {
                sessions.add(sessionId);
            }
```

Reply:
I did that earlier, but IntelliJ doesn't like this, even when it's a final variable. I'd rather not freak out every time I open the code than save two lines…